### DISCUSSION TITLE
Configuration Package Update Reconcile Length of Time

### DISCUSSION THREAD

**Discussion by natereid72** [crossplane/crossplane#2999] [Q&A]
Have noticed it taking 15-60 minutes for a change to a Configuration Package to reconcile in the cluster. Steps:

1. Install Configuration Package
2. Update Composition in Configuration Package and install new version
3. Wait 15-60 minutes for running Composition to show update

@negz pointed me to https://github.com/crossplane/crossplane/blob/master/internal/controller/apiextensions/composite/reconciler.go

Where it looks like the reconcile loop should be 2 minutes. Wondering why I'm seeing it take considerably longer?

**Comment by hasheddan** [comment]
@natereid72 does the new package revision become healthy in a reasonable period of time?

**Reply by natereid72** [reply]
Yep, pretty quickly actually. And the long reconcile times for Composition changes seem to vary, today they are 3 minutes or less.

**Reply by natereid72** [reply]
I'm going to test it for a week, and see if I can reproduce. When it was taking a long time, I was seeing the PackageRevision and Configuration package updating almost immediately. So it was a delay after that.

**Comment by negz** [comment]
@natereid72 Sorry, I misunderstood you when I linked to `internal/controller/apiextensions/composite/reconciler.go`. That handles _using_ the Composition to reconcile an XR with its composed resources (i.e. MRs). I believe this controller is the relevant one - it unpacks the Configuration package and reconciles its contents (e.g. Compositions) with what's in the Crossplane API: 

https://github.com/crossplane/crossplane/blob/25464654ca62411b581460818ae2e04399ff305d/internal/controller/pkg/revision/reconciler.go#L327
